models/equaler.go

```diff
@@ -0,0 +1,6 @@
+package models
+
+// The Equaler interface allows object that implement it, to be compared
+type Equaler interface {
+	Equal(Equaler) bool
```

Comment:
> If you implement Equal and the other one doesn't, then you're not Equal?

@aslakknutsen yes. But you're supposed to be calling Equal on the same types. So if the LHS has it but the RHS doesn't then you have a programming error.



> why not interface{} for the param type?

@tsmaeder Simply because the `Equaler` is the interface that has an `Equal` method. So we're at least assuring that the types to be compared are more closely related to each other than with pure `interface{}`.